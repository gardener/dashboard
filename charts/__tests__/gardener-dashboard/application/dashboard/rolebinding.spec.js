//
// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

'use strict'

const { helm } = fixtures

const renderTemplates = helm.renderDashboardApplicationTemplates

describe('gardener-dashboard', function () {
  describe('rolebinding', function () {
    let templates

    beforeEach(() => {
      templates = [
        'rolebinding'
      ]
    })

    it('should render the template with default values', async function () {
      const values = {}
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [roleBinding] = documents
      expect(roleBinding).toMatchSnapshot()
    })

    describe('when virtual garden is enabled', function () {
      it('should render the template', async function () {
        const values = {
          global: {
            virtualGarden: {
              enabled: true,
              userName: 'runtime-cluster:system:serviceaccount:garden:gardener-dashboard'
            },
            dashboard: {
              serviceAccountName: 'gardener-dashboard'
            }
          }
        }
        const documents = await renderTemplates(templates, values)
        expect(documents).toHaveLength(1)
        const [roleBinding] = documents
        expect(roleBinding).toMatchSnapshot()
      })
    })
  })
})
