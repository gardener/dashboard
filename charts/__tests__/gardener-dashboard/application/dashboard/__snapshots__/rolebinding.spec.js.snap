// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gardener-dashboard rolebinding should render the template with default values 1`] = `
{
  "apiVersion": "rbac.authorization.k8s.io/v1",
  "kind": "RoleBinding",
  "metadata": {
    "labels": {
      "app.kubernetes.io/component": "dashboard",
      "app.kubernetes.io/instance": "gardener-dashboard",
      "app.kubernetes.io/managed-by": "Helm",
      "app.kubernetes.io/name": "gardener-dashboard",
      "helm.sh/chart": "gardener-dashboard-application-0.1.0",
    },
    "name": "gardener.cloud:system:dashboard-github-webhook",
    "namespace": "garden",
  },
  "roleRef": {
    "apiGroup": "rbac.authorization.k8s.io",
    "kind": "Role",
    "name": "gardener.cloud:system:dashboard-github-webhook",
  },
  "subjects": [
    {
      "kind": "ServiceAccount",
      "name": "gardener-dashboard",
      "namespace": "garden",
    },
  ],
}
`;

exports[`gardener-dashboard rolebinding when virtual garden is enabled should render the template 1`] = `
{
  "apiVersion": "rbac.authorization.k8s.io/v1",
  "kind": "RoleBinding",
  "metadata": {
    "labels": {
      "app.kubernetes.io/component": "dashboard",
      "app.kubernetes.io/instance": "gardener-dashboard",
      "app.kubernetes.io/managed-by": "Helm",
      "app.kubernetes.io/name": "gardener-dashboard",
      "helm.sh/chart": "gardener-dashboard-application-0.1.0",
    },
    "name": "gardener.cloud:system:dashboard-github-webhook",
    "namespace": "garden",
  },
  "roleRef": {
    "apiGroup": "rbac.authorization.k8s.io",
    "kind": "Role",
    "name": "gardener.cloud:system:dashboard-github-webhook",
  },
  "subjects": [
    {
      "kind": "ServiceAccount",
      "name": "gardener-dashboard",
      "namespace": "garden",
    },
  ],
}
`;
